cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -498,16 +512,22 @@ public void getArtifactPlugin(HttpRequest request, HttpResponder responder,
                                 @PathParam("artifact-version") String artifactVersion,
                                 @PathParam("plugin-type") String pluginType,
                                 @PathParam("plugin-name") String pluginName,
-                                @QueryParam("scope") @DefaultValue("user") String scope)
-    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException {
+                                @QueryParam("scope") @DefaultValue("user") String scope,
+                                @QueryParam("versionRange") String versionRange,
+                                @QueryParam("limit") @DefaultValue("-1") String limit,
+                                @QueryParam("order") String order)
+    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException, InvalidArtifactRangeException {
 
     NamespaceId namespace = Ids.namespace(namespaceId);
     NamespaceId artifactNamespace = validateAndGetScopedNamespace(namespace, scope);
     Id.Artifact artifactId = validateAndGetArtifactId(artifactNamespace, artifactName, artifactVersion);
+    ArtifactRange range = versionRange == null ? null : ArtifactRange.parse(namespace, artifactName, versionRange);
```

Comment:
you should just be using a VersionRange here and not an ArtifactRange. This endpoint is used to get all plugins of a specific type and name that extend some parent. For example, all plugins of type 'batchsource' named 'Stream' that extend cdap-data-pipeline-4.2.0. The versionRange passed in the query params should be filtering based on the plugin artifact, but this range will contain the namespace and name of the parent. If you use ArtifactRange like this, you will filter out all plugins except those in the cdap-data-pipeline artifact, which will be nothing.